rust/tests/integration/webrtc_transport.rs

```diff
@@ -194,6 +195,29 @@ fn create_succeeds() {
                 }),
             );
             {
+                let ice_candidates = transport1.ice_candidates();
+                assert_eq!(ice_candidates.len(), 3);
+                assert_eq!(ice_candidates[0].ip, "9.9.9.1".parse::<IpAddr>().unwrap());
+                assert_eq!(ice_candidates[0].protocol, Protocol::Udp);
+                assert_eq!(ice_candidates[0].r#type, IceCandidateType::Host);
+                assert_eq!(ice_candidates[0].tcp_type, None);
+                assert_eq!(ice_candidates[1].ip, "9.9.9.2".parse::<IpAddr>().unwrap());
+                assert_eq!(ice_candidates[1].protocol, Protocol::Udp);
+                assert_eq!(ice_candidates[1].r#type, IceCandidateType::Host);
+                assert_eq!(ice_candidates[1].tcp_type, None);
+                assert_eq!(ice_candidates[2].ip, "127.0.0.1".parse::<IpAddr>().unwrap());
+                assert_eq!(ice_candidates[2].protocol, Protocol::Udp);
+                assert_eq!(ice_candidates[2].r#type, IceCandidateType::Host);
+                assert_eq!(ice_candidates[2].tcp_type, None);
+                assert_eq!(ice_candidates[2].ip, "127.0.0.1".parse::<IpAddr>().unwrap());
+                assert_eq!(ice_candidates[2].protocol, Protocol::Udp);
+                assert_eq!(ice_candidates[2].r#type, IceCandidateType::Host);
+                assert_eq!(ice_candidates[2].tcp_type, None);
+                assert!(ice_candidates[0].priority > ice_candidates[1].priority);
+                assert!(ice_candidates[1].priority > ice_candidates[2].priority);
+
+                // TODO: How comes UDP and TCP where created before?
```

Comment:
We were creating explicitly UDP sockets, how could we be expecting UDP and TCP? cc @nazar-pc.

Reply:
IIRC `ListenInfo` is a relatively new thing, apparerntly something was off there? Check git history for this section of the code, I do not remember off top of my head.

Reply:
What do we mean here? Is the test expecting a TCP socket even if only UDP was requested? Is this a bug in Rust create_webrtc_transport? In Rust webrtc_transport constructor? Or just in the test?

Reply:
If test was passing before worker must have been creating TCP socket in the past as well. I don't think Rust side was hallucinating things that were not there. Maybe it was fixed with flatbuffers update? IDK.

Reply:
I migrated `ListenIps` to `ListenInfos` in Rust entirely (while in Node we still allow `listenIps` for backwards compatibility) so it's imposible that Rust side is creating TCP sockets if not requested by the app.

Reply:
Wow, something must be wrong since the test is indeed just creating UDP sockets:

```rust
let transport1 = router
                .create_webrtc_transport({
                    let mut webrtc_transport_options = WebRtcTransportOptions::new(
                        vec![
                            ListenInfo {
                                protocol: Protocol::Udp,
                                ip: IpAddr::V4(Ipv4Addr::LOCALHOST),
                                announced_ip: Some("9.9.9.1".parse().unwrap()),
                                port: None,
                                send_buffer_size: None,
                                recv_buffer_size: None,
                            },
                            ListenInfo {
                                protocol: Protocol::Udp,
                                ip: IpAddr::V4(Ipv4Addr::UNSPECIFIED),
                                announced_ip: Some("9.9.9.2".parse().unwrap()),
                                port: None,
                                send_buffer_size: None,
                                recv_buffer_size: None,
                            },
                            ListenInfo {
                                protocol: Protocol::Udp,
                                ip: IpAddr::V4(Ipv4Addr::LOCALHOST),
                                announced_ip: None,
                                port: None,
                                send_buffer_size: None,
                                recv_buffer_size: None,
                            },
                        ]
                        .try_into()
                        .unwrap(),
                    );
                    webrtc_transport_options.num_sctp_streams = NumSctpStreams {
                        os: 2048,
                        mis: 2048,
                    };
                    webrtc_transport_options.max_sctp_message_size = 1000000;
                    webrtc_transport_options.app_data = AppData::new(CustomAppData { foo: "bar" });

                    webrtc_transport_options
                })
```

Reply:
Check commented-out code below

Reply:
Oh hell, sorry.

Reply:
So the test passes in `v3` and `flatbuffers` branch because:

```
#[test]
#[ignore]
```

¯\_(ツ)_/¯

@jmillan I'm trying to fix it in this branch directly.

Reply:
Guys, there is nothing to test. The test was disabled in `v3` and `flatbuffers` branches, and in this branch it is enabled, fixed and it passes. That's all.

@jmillan I'll remove the commented-out code and push after confirming everything is ok.

Reply:
Pushed. Resolving this thread.